### PR TITLE
Add button for downloading digital book specimens

### PR DESCRIPTION
**Pull request by yohanboniface**

### CODE REVIEW

File: ideascube/library/models.py [hunk 1]
```diff
@@ -99,3 +101,10 @@ def __str__(self):
 
     def get_absolute_url(self):
         return reverse('library:book_detail', kwargs={'pk': self.item.pk})
+
+    @property
+    def extension(self):
+        if not self.file:
+            return ''
+        name, extension = os.path.splitext(self.file.name)
+        return extension[1:]
```
Comment (vinyll) [inline ideascube/library/models.py]: for not so basic extension (like _.tar.gz_) it may not be reliable.

Maybe something like:
```
file_name = self.file.name
extension = file_name[file_name.find('.'):]
```

File: ideascube/library/tests/test_models.py [hunk 2]
```diff
@@ -79,3 +79,18 @@ def test_unicode_returns_digital_specimen_of_book():
     book = BookFactory()
     specimen = DigitalBookSpecimenFactory(item=book)
     assert str(specimen).startswith(u'Digital specimen of')
+
+
+def test_specimen_extension():
+    specimen = DigitalBookSpecimenFactory(file__filename='book.epub')
+    assert specimen.extension == 'epub'
```
Comment (vinyll) [inline ideascube/library/tests/test_models.py]: add a test for double dotted extensions

File: ideascube/library/models.py (hunk 1 above)
Comment (yohanboniface) [inline ideascube/library/models.py]: Agreed that there are edge cases, but I think whatever the split method we chose we have some.
That said I think we can stay on that simple option, that should cover the expected extension ('.epub', '.pdf'…), and if someone actually upload a complex extension, that will not be such an issue.

Comment (yohanboniface) [inline ideascube/library/models.py]: Agreed that there are edge cases, but I think whatever the split method we chose we have some.
That said I think we can stay on that simple option, that should cover the expected extension ('.epub', '.pdf'…), and if someone actually upload a complex extension, that will not be such an issue.
